ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Worker.java

```diff
@@ -291,6 +299,18 @@ protected Boolean findNextCompactionAndExecute(boolean collectGenericStats, bool
         return false;
       }
 
+      if (ci.type.equals(CompactionType.REBALANCE) && t1.getSd().isSetNumBuckets() && t1.getSd().getNumBuckets() > 0) {
+        LOG.error("Cannot execute rebalancing compaction on bucketed tables.");
+        ci.errorMessage = "Cannot execute rebalancing compaction on bucketed tables.";
+        msc.markRefused(CompactionInfo.compactionInfoToStruct(ci));
+        return false;
+      }
+
+      if (!ci.type.equals(CompactionType.REBALANCE) && ci.numberOfBuckets > 0) {
+        LOG.warn("Only the REBALANCE compaction accepts the number of buckets clause (CLUSTERED INTO {N} BUCKETS). " +
+            "Since the compaction request is " + ci.type + ", it will be ignored.");
```

Comment:
Please use placeholders in log messages instead of string concatenation like
```
LOG.warn("Only the REBALANCE compaction accepts the number of buckets clause (CLUSTERED INTO {N} BUCKETS). " +
            "Since the compaction request is {}, it will be ignored.", ci.type);
```